Services/CO.CDP.Organisation.WebApi/UseCase/UpdateOrganisationUseCase.cs

```diff
@@ -116,4 +123,35 @@ public async Task<bool> Execute((Guid organisationId, UpdateOrganisation updateO
 
         return await Task.FromResult(true);
     }
+
+    private void RemoveIdentifier(OrganisationInformation.Persistence.Organisation organisation,
+        OrganisationInformation.Persistence.Organisation.Identifier identifierToRemove)
+    {
+        organisation.Identifiers.Remove(identifierToRemove);
+
+        var nextPrimaryIdentifier = organisation.Identifiers.FirstOrDefault(i =>
+            i.Scheme != AssignIdentifierUseCase.IdentifierSchemes.Ppon &&
+            i.Scheme != AssignIdentifierUseCase.IdentifierSchemes.Other);
+
+        if (nextPrimaryIdentifier == null)
+        {
+            nextPrimaryIdentifier = organisation.Identifiers.FirstOrDefault(i =>
+                i.Scheme == AssignIdentifierUseCase.IdentifierSchemes.Ppon);
+
+            if (nextPrimaryIdentifier == null)
+            {
+                nextPrimaryIdentifier = organisation.Identifiers.FirstOrDefault(i =>
+                    i.Scheme == AssignIdentifierUseCase.IdentifierSchemes.Other);
+            }
+        }
```

Comment:
I'm not sure about this code. It only seems to be looking at PPON and Other type of identifiers. Doesn't seem to be right.

Reply:
Now checks whether the primary identifier is being removed. If it is, it finds the next primary identifier, if there is one.